features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml

```diff
@@ -609,6 +609,9 @@
         <!--GroupEPUrl>${carbon.protocol}://${carbon.host}:${carbon.management.port}/scim2/Groups</GroupEPUrl-->
         <!--<ComplexMultiValuedAttributeSupportEnabled>true</ComplexMultiValuedAttributeSupportEnabled>-->
         <!--<EnableFilteringEnhancements>true</EnableFilteringEnhancements>-->
+        <!--<FilterUsersAndGroupsOnlyFromPrimaryDomain>true</FilterUsersAndGroupsOnlyFromPrimaryDomain>-->
+        <MandateDomainForGroupNamesInGroupsResponse>true</MandateDomainForGroupNamesInGroupsResponse>
```

Comment:
I don't think this needs be true by default. It results on inconsistent group display names in responses.
Let's keep the default behavior false. Also, IMO, the config commented also should by default include the default value that should be engaged

Reply:
Fixed